src/averageAges.js

```diff
@@ -20,28 +20,64 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+
+  const menLength = people.filter(person => person.sex === 'm').length;
+
+  const menCenturyLength = people.filter(person =>
+    person.sex === 'm' && Math.ceil(person.died / 100) === century).length;
+
+  const menAgeSum = people.reduce((acc, person) =>
+    person.sex === 'm' ? acc + (person.died - person.born) : acc + 0, 0);
+
+  const menAverage = menAgeSum / menLength;
+
+  const menAgeCentury = people.reduce((acc, person) =>
+    person.sex === 'm' && Math.ceil(person.died / 100) === century
+      ? acc + (person.died - person.born) : acc + 0, 0);
+
+  const menAverageCentury = menAgeCentury / menCenturyLength;
+
+  return century === undefined ? menAverage : menAverageCentury;
```

Comment:
```suggestion
  return century ? menAverageCentury : menAverage ;
```
Наврят ли кто станет вбрасывать в century 0, так что это должно сработать:) И выглядит намного читабельнее